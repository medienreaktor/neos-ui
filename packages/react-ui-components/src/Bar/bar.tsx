import mergeClassNames from 'classnames';
import React, {PureComponent} from 'react';

type BarPosition = 'top' | 'bottom';

interface BarTheme {
    readonly 'bar': string;
    readonly 'bar--top': string;
    readonly 'bar--bottom': string;
}

export interface BarProps {
    /**
     * This prop controls the vertical positioning of the Bar.
     */
    readonly position: BarPosition;

    /**
     * An optional css theme to be injected.
     */
    readonly theme?: BarTheme;

    /**
     * An optional `className` to attach to the wrapper.
     */
    readonly className?: string;

    /**
     * The contents to be rendered within the `Bar`.
     */
    readonly children: React.ReactNode;
}

class Bar extends PureComponent<BarProps> {
    public render(): JSX.Element {
        const {position, className, theme, children, ...rest} = this.props;
        const finalClassName = mergeClassNames(
            className,
            theme!.bar,
            {
                [theme!['bar--top']]: position === 'top',
                [theme!['bar--bottom']]: position === 'bottom'
            }
        );

        // TODO:
        //  The prop 'rest' is empty here. I suppose it contains all the attributes that are not part of the Badge Api/Interface.
        //  Consider enxtending Bagde with React.HTMLAttributes<HTMLDivElement> like we did in the Button component.
        //  It also enables autocomplete for the whole div Api.
        //  Consider this for the other components.
        return (
            <div className={finalClassName} {...rest}>
                {children}
            </div>
        );
    }
}

export default Bar;
