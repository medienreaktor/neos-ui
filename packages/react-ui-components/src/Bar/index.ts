import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from '../identifiers';

import Bar from './bar';
import style from './style.css';

export default themr(identifiers.bar, style)(Bar);
