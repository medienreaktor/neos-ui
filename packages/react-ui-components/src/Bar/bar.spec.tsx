import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Bar, {BarProps} from './bar';

describe('<Bar/>', () => {
    const props: BarProps = {
        children: 'Foo children',
        position: 'top',
        theme: {
            'bar': 'barClassName',
            'bar--bottom': 'bottomClassName',
            'bar--top': 'topClassName',
        }
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<Bar {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<Bar {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });
});
