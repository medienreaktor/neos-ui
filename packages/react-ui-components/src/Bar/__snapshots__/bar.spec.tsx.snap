// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar/> should render correctly. 1`] = `
<div
  className="barClassName topClassName"
>
  Foo children
</div>
`;
